src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CodeGenerator.cs

```diff
@@ -38,7 +38,7 @@ private static MethodInfo ObjectEquals
             {
                 if (s_objectEquals == null)
                 {
-                    s_objectEquals = Globals.TypeOfObject.GetMethod("Equals", BindingFlags.Public | BindingFlags.Static);
+                    s_objectEquals = typeof(object).GetMethod("Equals", BindingFlags.Public | BindingFlags.Static);
```

Comment:
It would be nice to delete `Globals.TypeOfObject` and all other typeof caches from the `Globals` type.

Reply:
Agreed. I've opened https://github.com/dotnet/runtime/issues/42861 to do this.